backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/acl/service/AclManagerService.kt

```diff
@@ -0,0 +1,121 @@
+package com.tngtech.apicenter.backend.connector.acl.service
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import org.springframework.security.acls.domain.ObjectIdentityImpl
+import org.springframework.security.acls.model.*
+import org.springframework.transaction.TransactionStatus
+import org.springframework.transaction.support.AbstractPlatformTransactionManager
+import org.springframework.transaction.support.TransactionCallback
+import java.io.Serializable
+import org.springframework.transaction.support.TransactionCallbackWithoutResult
+import org.springframework.transaction.support.TransactionTemplate
+import org.springframework.security.acls.model.MutableAcl
+import org.springframework.security.acls.model.ObjectIdentity
+import mu.KotlinLogging
+import org.springframework.security.acls.domain.BasePermission
+import java.lang.IllegalArgumentException
+
+private val logger = KotlinLogging.logger {  }
+
+interface AclManager {
+    fun <T> addPermission(domainClass: Class<T>, id: Serializable, sid: Sid, permission: Permission)
+    fun <T> removePermission(domainClass: Class<T>, id: Serializable, sid: Sid, permission: Permission)
+    fun <T> hasPermission(domainClass: Class<T>, id: Serializable, sid: Sid, permission: Permission): Boolean
+    fun <T> clearPermissions(domainClass: Class<T>, id: Serializable)
+}
+
+@Service
+class SpecificationPermissionManager @Autowired constructor(private val aclManagerService : AclManagerService) {
+    fun addPermission(id: Long, sid: Sid, permission: Permission) {
+        logger.info("Add ${permissionToString(permission)} permission to spec $id ")
+        return aclManagerService.addPermission(Long::class.java, id, sid, permission)
+    }
+
+    fun removePermission(id: Long, sid: Sid, permission: Permission) {
+        logger.info("Remove ${permissionToString(permission)} permission from spec $id ")
+        return aclManagerService.removePermission(Long::class.java, id, sid, permission)
+    }
+
+    fun hasPermission(id: Long, sid: Sid, permission: Permission): Boolean {
```

Comment:
To be honest, I don't understand what `sid` is and why it's different from `id`. Could we please rename it?